### ISSUE TITLE
Rancher pod shows [FATAL] invalid character '-' after object key:value pair

### ISSUE THREAD

**Issue by QuintonMcLeod** [rancher/rancher#36416]
So, I've been using Rancher for about a year now on microk8s. Worked great. Suddenly, when logging into Rancher, I'd get the "invalid character '-' after object key:value pair" error in the browser. No matter how many times I tried to login, I'd get the same error. So, I figured I'd remove Rancher from Microk8s and reinstall, right? Wrong!

So, I use Helm charts to install a lot of my software on Kubernetes. So, I used Rancher's official Helm chart, uninstalled and then reinstalled. Goes through without a hitch.... Except, the pods keep crashing. I check the log on one of the pods and this is what I see:

------------------------------------------------------------------------------------------------------------------------------------------------
2022/02/06 04:27:46 [INFO] Rancher version v2.6.3 (3c1d5fac3) is starting
2022/02/06 04:27:46 [INFO] Rancher arguments {ACMEDomains:[] AddLocal:true Embedded:false BindHost: HTTPListenPort:80 HTTPSListenPort:443 K8sMode:auto Debug:false Trace:false NoCACerts:false AuditLogPath:/var/log/auditlog/rancher-api-audit.log AuditLogMaxage:10 AuditLogMaxsize:100 AuditLogMaxbackup:10 AuditLevel:0 Features: ClusterRegistry:}
2022/02/06 04:27:46 [INFO] Listening on /tmp/log.sock
2022/02/06 04:27:47 [INFO] Running in clustered mode with ID 10.1.74.120, monitoring endpoint cattle-system/rancher
2022/02/06 04:27:48 [INFO] Applying CRD features.management.cattle.io
2022/02/06 04:27:53 [INFO] Applying CRD navlinks.ui.cattle.io
2022/02/06 04:27:53 [INFO] Applying CRD clusters.management.cattle.io
2022/02/06 04:27:54 [INFO] Applying CRD apiservices.management.cattle.io
2022/02/06 04:27:54 [INFO] Applying CRD clusterregistrationtokens.management.cattle.io
2022/02/06 04:27:54 [INFO] Applying CRD settings.management.cattle.io
2022/02/06 04:27:55 [INFO] Applying CRD preferences.management.cattle.io
2022/02/06 04:27:55 [INFO] Applying CRD features.management.cattle.io
2022/02/06 04:27:56 [INFO] Applying CRD clusterrepos.catalog.cattle.io
2022/02/06 04:27:56 [INFO] Applying CRD operations.catalog.cattle.io
2022/02/06 04:27:56 [INFO] Applying CRD apps.catalog.cattle.io
2022/02/06 04:27:57 [INFO] Applying CRD fleetworkspaces.management.cattle.io
2022/02/06 04:27:57 [INFO] Applying CRD managedcharts.management.cattle.io
2022/02/06 04:27:57 [INFO] Applying CRD clusters.provisioning.cattle.io
2022/02/06 04:27:58 [INFO] Applying CRD clusters.provisioning.cattle.io
2022/02/06 04:27:58 [INFO] Applying CRD rkeclusters.rke.cattle.io
2022/02/06 04:27:59 [INFO] Applying CRD rkecontrolplanes.rke.cattle.io
2022/02/06 04:27:59 [INFO] Applying CRD rkebootstraps.rke.cattle.io
2022/02/06 04:27:59 [INFO] Applying CRD rkebootstraptemplates.rke.cattle.io
2022/02/06 04:28:00 [INFO] Applying CRD rkecontrolplanes.rke.cattle.io
2022/02/06 04:28:00 [INFO] Applying CRD custommachines.rke.cattle.io
2022/02/06 04:28:00 [INFO] Applying CRD clusters.cluster.x-k8s.io
2022/02/06 04:28:01 [INFO] Applying CRD machinedeployments.cluster.x-k8s.io
2022/02/06 04:28:01 [INFO] Applying CRD machinehealthchecks.cluster.x-k8s.io
2022/02/06 04:28:02 [INFO] Applying CRD machines.cluster.x-k8s.io
2022/02/06 04:28:03 [INFO] Applying CRD machinesets.cluster.x-k8s.io
2022/02/06 04:28:12 [FATAL] invalid character '-' after object key:value pair`
---------------------------------------------------------------------------------------------------------------------------------------------

Then the pod crashes and restarts over and over and over again. 

I have no idea what this error even means or where to go to attempt to fix it. It's just all around strange.

**Comment by mhsraft** [comment]
I got the same error

54 reflector.go:533] pkg/mod/github.com/rancher/client-go@v1.27.4-rancher1/tools/cache/reflector.go:231: failed to ││  list *summary.SummarizedObject: invalid character 'p' after object key

**Comment by QuintonMcLeod** [comment]
> I got the same error
> 
> 54 reflector.go:533] pkg/mod/github.com/rancher/client-go@v1.27.4-rancher1/tools/cache/reflector.go:231: failed to ││ list *summary.SummarizedObject: invalid character 'p' after object key

This is not the same error.

Anyway, I fixed it by completely Uninstaller and purging Microk8s. Afterward, I reinstalled it and then put Rancher on it. Worked like a charm.

Reality is, I had Microk8s installed on baremetal. That's bad practice, as Kubernetes acts weird and often times needs a good system reboot to act right. So, I ended up installing it on a VM. Once I started doing that, it became very managable.
